lib/ddtrace/contrib/mongodb/subscribers.rb

```diff
@@ -0,0 +1,104 @@
+          # https://github.com/mongodb/mongo-ruby-driver/blob/master/lib/mongo/monitoring.rb#L70
+          # https://github.com/mongodb/mongo-ruby-driver/blob/master/lib/mongo/monitoring/publishable.rb#L38-L56
+          span = pin.tracer.trace('mongo.cmd', service: pin.service, span_type: Datadog::Ext::Mongo::TYPE)
+          Thread.current[:datadog_mongo_span] = span
+
+          # common fields for all commands
+          command_name = event.command_name
+          collection = event.command[command_name]
+          span.set_tag(Datadog::Ext::Mongo::COLLECTION, collection)
+          span.set_tag(Datadog::Ext::Mongo::DB, event.database_name)
+          span.set_tag(Datadog::Ext::NET::TARGET_HOST, event.address.host)
+          span.set_tag(Datadog::Ext::NET::TARGET_PORT, event.address.port)
+
+          # commands are handled so that specific fields are normalized based on type, if it's
+          # a command that requires documents or a specific query. For some commands, we only
+          # take in consideration the query ('q') to keep the cardinality low
+          # NOTE: 'find' doesn't use a symbol
+          case command_name
+          when 'find'
+            filter = event.command['filter']
+            unless filter.nil? || filter.empty?
+              query = Datadog::Contrib::MongoDB.normalize_query(filter)
+              span.set_tag('mongodb.filter', query)
+            end
+          when :update
+            updates = event.command['updates']
+            unless updates.nil? || updates.empty? || updates.first['q'].empty?
+              query = Datadog::Contrib::MongoDB.normalize_query(updates.first['q'])
+              span.set_tag('mongodb.updates', query)
+            end
+          when :delete
+            deletes = event.command['deletes']
+            unless deletes.nil? || deletes.empty? || deletes.first['q'].empty?
+              query = Datadog::Contrib::MongoDB.normalize_query(deletes.first['q'])
+              span.set_tag('mongodb.deletes', query)
+            end
+          else
+            span.set_tag('mongodb.ordered', event.command['ordered'])
+
+            documents = event.command['documents']
+            unless documents.nil? || documents.empty?
+              document = Datadog::Contrib::MongoDB.normalize_documents(documents)
+              span.set_tag('mongodb.documents', document)
+            end
+          end
+
+          # set the resource with the quantized documents or queries
+          span.resource = "#{command_name} #{collection} #{document || query}".strip
+        end
+
+        def failed(event)
+          span = Thread.current[:datadog_mongo_span]
+          return unless span
+
+          # the failure is not a real exception because it's handled by
+          # the framework itself, so setting the error and the message
+          # should be enough
+          span.status = 1
+          span.set_tag(Datadog::Ext::Errors::MSG, event.message)
```

Comment:
Here I'm not storing the full `Exception`, and it should be enough I think

Reply:
@palazzem, could you please use the `span#set_error` for that?

Reply:
oh of course! now it's available in our master! thanks for pointing that!

Reply:
since this is a Mongo event and not an exception, I thought the `set_error()` couldn't handle that, but I see you return a `BlankError` (that is an `Error` instance). Let me make this change so.